### PR TITLE
Support for IBM Open Platform

### PR DESCRIPTION
**Pull request by wolf31o2**
This has been tested on a cluster using IOP 4.1.0.0 using chef-solo.

A list of the software enclosed in IOP is available on [IBM's site](http://www-01.ibm.com/support/knowledgecenter/api/content/nl/en-us/SSPT3X_4.1.0/com.ibm.swg.im.infosphere.biginsights.install.doc/doc/c0057609.html).

### CODE REVIEW

**Comment by wolf31o2** [comment]
While testing this, I ran into issues installing `hadoop-client`...

```
# yum list available | grep hadoop
hadoop-lzo.x86_64                          0.5.1-1                     IOP-UTILS-1.1.0.0
hadoop-lzo-native.x86_64                   0.5.1-1                     IOP-UTILS-1.1.0.0
hadoop_4_1_0_0.x86_64                      2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-client.x86_64               2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-conf-pseudo.x86_64          2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-debuginfo.x86_64            2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-doc.x86_64                  2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-hdfs.x86_64                 2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-hdfs-datanode.x86_64        2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-hdfs-fuse.x86_64            2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-hdfs-journalnode.x86_64     2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-hdfs-namenode.x86_64        2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-hdfs-secondarynamenode.x86_64
hadoop_4_1_0_0-hdfs-zkfc.x86_64            2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-httpfs.x86_64               2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-libhdfs.x86_64              2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-libhdfs-devel.x86_64        2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-mapreduce.x86_64            2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-mapreduce-historyserver.x86_64
hadoop_4_1_0_0-yarn.x86_64                 2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-yarn-nodemanager.x86_64     2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-yarn-proxyserver.x86_64     2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x
hadoop_4_1_0_0-yarn-resourcemanager.x86_64 2.7.1_IBM_8.4.1.0.0-3.el6   IOP-4.1.x

# yum install hadoop-client
Loaded plugins: fastestmirror
Setting up Install Process
Loading mirror speeds from cached hostfile
 * base: centos.firehosted.com
 * extras: mirrors.lga7.us.voxel.net
 * updates: mirror.net.cen.ct.gov
No package hadoop-client available.
Error: Nothing to do
```

This will need conditional code on the package names to be added to supporting recipes.

File: attributes/default.rb [hunk 1]
```diff
@@ -124,6 +126,14 @@
...
+  default['hadoop']['mapred_site']['mapreduce.admin.map.child.java.opts'] = '-server -Djava.net.preferIPv4Stack=true -Diop.version=${iop.version}'
+  default['hadoop']['mapred_site']['mapreduce.admin.user.env'] = 'LD_LIBRARY_PATH=/usr/iop/${iop.version}/hadoop/lib/native'
+  default['hadoop']['mapred_site']['mapreduce.application.framework.path'] = '/iop/apps/${iop.version}/mapreduce/mapreduce.tar.gz#mr-framework'
+  default['hadoop']['mapred_site']['mapreduce.application.classpath'] = '$PWD/mr-framework/hadoop/share/hadoop/mapreduce/*:$PWD/mr-framework/hadoop/share/hadoop/mapreduce/lib/*:$PWD/mr-framework/hadoop/share/hadoop/common/*:$PWD/mr-framework/hadoop/share/hadoop/common/lib/*:$PWD/mr-framework/hadoop/share/hadoop/yarn/*:$PWD/mr-framework/hadoop/share/hadoop/yarn/lib/*:$PWD/mr-framework/hadoop/share/hadoop/hdfs/*:$PWD/mr-framework/hadoop/share/hadoop/hdfs/lib/*:/usr/iop/${iop.version}/hadoop/lib/hadoop-lzo-0.5.1.jar:/etc/hadoop/conf/secure'
+  default['hadoop']['mapred_site']['yarn.app.mapreduce.am.admin-command-opts'] = '-Diop.version=${iop.version}'
```
Comment (wolf31o2) [inline attributes/default.rb]: This fixed MR...
```
[chris@iop7287-1000 ~]$ yarn jar /usr/iop/current/hadoop-mapreduce-client/hadoop-mapreduce-examples.jar randomwriter -Dtest.randomwrite.total_bytes=10000000 /tmp/test-after-upgrade
16/04/07 19:09:29 INFO client.RMProxy: Connecting to ResourceManager at iop7287-1000.dev.continuuity.net/10.240.0.24:8032
Running 10 maps.
Job started: Thu Apr 07 19:09:30 UTC 2016
16/04/07 19:09:30 INFO client.RMProxy: Connecting to ResourceManager at iop7287-1000.dev.continuuity.net/10.240.0.24:8032
16/04/07 19:09:30 INFO mapreduce.JobSubmitter: number of splits:10
16/04/07 19:09:30 INFO mapreduce.JobSubmitter: Submitting tokens for job: job_1460054007763_0003
16/04/07 19:09:31 INFO impl.YarnClientImpl: Submitted application application_1460054007763_0003
16/04/07 19:09:31 INFO mapreduce.Job: The url to track the job: http://REDACTED:8088/proxy/application_1460054007763_0003/
16/04/07 19:09:31 INFO mapreduce.Job: Running job: job_1460054007763_0003
16/04/07 19:09:37 INFO mapreduce.Job: Job job_1460054007763_0003 running in uber mode : false
16/04/07 19:09:37 INFO mapreduce.Job:  map 0% reduce 0%
16/04/07 19:10:37 INFO mapreduce.Job:  map 10% reduce 0%
16/04/07 19:10:41 INFO mapreduce.Job:  map 40% reduce 0%
16/04/07 19:10:45 INFO mapreduce.Job:  map 60% reduce 0%
16/04/07 19:11:27 INFO mapreduce.Job:  map 70% reduce 0%
16/04/07 19:11:31 INFO mapreduce.Job:  map 100% reduce 0%
16/04/07 19:11:35 INFO mapreduce.Job: Job job_1460054007763_0003 completed successfully
16/04/07 19:11:35 INFO mapreduce.Job: Counters: 33
	File System Counters
		FILE: Number of bytes read=0
		FILE: Number of bytes written=1166390
		FILE: Number of read operations=0
		FILE: Number of large read operations=0
		FILE: Number of write operations=0
		HDFS: Number of bytes read=1410
		HDFS: Number of bytes written=10772829072
		HDFS: Number of read operations=40
		HDFS: Number of large read operations=0
		HDFS: Number of write operations=20
	Job Counters
		Killed map tasks=3
		Launched map tasks=13
		Other local map tasks=13
		Total time spent by all maps in occupied slots (ms)=2401872
		Total time spent by all reduces in occupied slots (ms)=0
		Total time spent by all map tasks (ms)=600468
		Total vcore-seconds taken by all map tasks=600468
		Total megabyte-seconds taken by all map tasks=614879232
	Map-Reduce Framework
		Map input records=10
		Map output records=1021811
		Input split bytes=1410
		Spilled Records=0
		Failed Shuffles=0
		Merged Map outputs=0
		GC time elapsed (ms)=2407
		CPU time spent (ms)=260530
		Physical memory (bytes) snapshot=2077048832
		Virtual memory (bytes) snapshot=8987316224
		Total committed heap usage (bytes)=2487222272
	org.apache.hadoop.examples.RandomWriter$Counters
		BYTES_WRITTEN=10737476956
		RECORDS_WRITTEN=1021811
	File Input Format Counters
		Bytes Read=0
	File Output Format Counters
		Bytes Written=10772829072
Job ended: Thu Apr 07 19:11:35 UTC 2016
The job took 125 seconds.
```

**Comment by wolf31o2** [comment]
This is now ready for review.

File: recipes/_compression_libs.rb [hunk 2]
```diff
@@ -37,6 +37,8 @@
   when 'rhel'
     pkgs += ['lzo', 'lzo-devel', 'hadooplzo', 'hadooplzo-native']
   end
+elsif iop?
+  pkgs += ['lzo', 'lzo-devel', 'hadoop-lzo', 'hadoop-lzo-native']
```
Comment (awholegunch) [inline recipes/_compression_libs.rb]: These packages right? Not as per line 38?
```
'hadooplzo', 'hadooplzo-native'

**Comment by awholegunch** [comment]
One minor question, otherwise LGTM.

File: recipes/_compression_libs.rb (hunk 2 above)
Comment (wolf31o2) [inline recipes/_compression_libs.rb]: Correct.

screen shot 2016-09-08 at 2 19 45 pm

**Pull request merged**